### PR TITLE
add pv_system_row_number

### LINKED ISSUE #302: Check that `PVDataSource.get_example()` correctly chooses PV systems.  `pv_system_row_number` is assigned to but never used.
## Detailed Description
In `PVDataSource.get_example()`, `pv_system_row_number` is assigned to but never used.  That could be a bug???

### PR DESCRIPTION
**Pull request by peterdudfield**
# Pull Request

## Description

- add pv_system_row_number to pv data set
- added it to fake data
- added a validation int he model to check it is there

Fixes #302 

## How Has This Been Tested?

normal unittests

- [ ] No
- [x] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: nowcasting_dataset/data_sources/fake.py [hunk 1]
```diff
@@ -238,6 +238,7 @@ def create_gsp_pv_dataset(
 
     data["x_coords"] = x_coords
     data["y_coords"] = y_coords
+    data["pv_system_row_number"] = data["id"] + 1000
```
Comment (jacobbieker) [inline nowcasting_dataset/data_sources/fake.py:241]: Why is 1000 added here?

**Review by jacobbieker — APPROVED** [review]
LGTM! Just one minor comment

File: nowcasting_dataset/data_sources/fake.py (hunk 1 above)
Comment (peterdudfield) [reply nowcasting_dataset/data_sources/fake.py:241]: just wanted it a bit different that the row number, could generate random numbers instead

Comment (jacobbieker) [reply nowcasting_dataset/data_sources/fake.py:241]: Ah okay, could we add a comment explaining it then?

**Pull request merged**